# Pandas

Pandas merupakan sebuah package sebagai alat analisis dan struktur untuk bahasa pemrograman python. Dengan menggunakan pandas kita dapat mengolah data dengan mudah, salah satu fiturnya adalah dataframe. Dengan fitur dataframe kita dapat membaca sebuah file dan menjadikan tabel. Selain itu, salah satu kegunaan pandas yaitu dapat membaca file dengan bentuk .txt,.csv, .xlsx, dll. 

Pertama, panggil package pandas sebagai berikut

In [1]:
import pandas as pd

Inputkan data_nilai_matematika dalam bentuk excel

In [2]:
data = pd.read_excel('data/data_nilai_matematika.xlsx')

Cetak data dengan menggunakan perintah print

In [3]:
print(data)

      Nama    Kelamin  Nilai
0     Andi  Laki-laki     80
1      Iva  Perempuan     74
2    Ihsan  Laki-laki     75
3     Dewi  Perempuan     76
4   Kurnia  Perempuan     39
5      Eko  Laki-laki     65
6     Tika  Perempuan     73
7    Fitri  Perempuan     60
8     Gita  Perempuan     63
9   Hafidz  Laki-laki     75
10    Maya  Perempuan     69
11  Lukman  Laki-laki     75
12   April  Perempuan     90
13     Eka  Perempuan     84
14    Wati  Perempuan     76


Dari data diatas diketahui bahwa data terdiri dari tiga variabel yaitu Nama, Kelamin dan NIlai. 

Menampilkan 5 data pertama 

In [4]:
print(data.head())

     Nama    Kelamin  Nilai
0    Andi  Laki-laki     80
1     Iva  Perempuan     74
2   Ihsan  Laki-laki     75
3    Dewi  Perempuan     76
4  Kurnia  Perempuan     39


Menampilkan 2 data pertama

In [5]:
print(data.head(2))

   Nama    Kelamin  Nilai
0  Andi  Laki-laki     80
1   Iva  Perempuan     74


Menampilkan data ke-3 sampai ke-8

In [6]:
print(data[3:9])

     Nama    Kelamin  Nilai
3    Dewi  Perempuan     76
4  Kurnia  Perempuan     39
5     Eko  Laki-laki     65
6    Tika  Perempuan     73
7   Fitri  Perempuan     60
8    Gita  Perempuan     63


Melihat info data

In [7]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 3 columns):
Nama       15 non-null object
Kelamin    15 non-null object
Nilai      15 non-null int64
dtypes: int64(1), object(2)
memory usage: 440.0+ bytes
None


Dari output diatas didapatkan jumlah baris data sebanyak 15 dengan variabel Nama, Kelamin dan Nilai juga sebnayak 15. Variabel Nama dan Kelamin bertipe data objek dan Nilai bertipe data integer. 

Baik, sebelumnya kita akan membahas tipe data yang sering digunakan
- Objek : Tipe data huruf, simbol,dll
- Integer : Tipe data untuk data numerik bilangan bulat
- Float : Tipe data untuk data numerik bilangan pecahan dan bisa juga bilangan bulat (biasanya untuk data pengukuran)
- Boolean : Tipe data dengan keluaran True atau False

Melihat deskripsi data

In [8]:
print(data.describe())

           Nilai
count  15.000000
mean   71.600000
std    11.849051
min    39.000000
25%    67.000000
50%    75.000000
75%    76.000000
max    90.000000


Dari output diatas merupakan deskripsi data dengan jumlah variabel Nilai sebanyak 15 data, rata-rata (mean) sebesar 71.6, standar deviasi (std) sebesar 11.84, nilai terkecil (min) sebesar 39, quartil pertama (25%) sebesar 67, quartil kedua atau median (50%) sebesar 75, quartil ketiga (75%) sebesar 76 dan nilai tertinggi (max) sebesar 90. 

Mengurutkan data sesuai variabel "Nama"

In [9]:
data = data.sort_values('Nama')
data

Nama    Kelamin  Nilai
0     Andi  Laki-laki     80
12   April  Perempuan     90
3     Dewi  Perempuan     76
13     Eka  Perempuan     84
5      Eko  Laki-laki     65
7    Fitri  Perempuan     60
8     Gita  Perempuan     63
9   Hafidz  Laki-laki     75
2    Ihsan  Laki-laki     75
1      Iva  Perempuan     74
4   Kurnia  Perempuan     39
11  Lukman  Laki-laki     75
10    Maya  Perempuan     69
6     Tika  Perempuan     73
14    Wati  Perempuan     76

Dapat dilihat dari tabel diatas bahwa data terurut sesuai dengan variabel "Nama" yang diurutkan dari abjad A-Z. Variabel "Kelamin" dan "Nilai" akan menyesuaikan pengurutannya. 

Mengurutkan nomor data yang telah di urutkan sesuai "Name" sebelumnya 

In [10]:
data = data.reset_index(drop=True)
print(data)

      Nama    Kelamin  Nilai
0     Andi  Laki-laki     80
1    April  Perempuan     90
2     Dewi  Perempuan     76
3      Eka  Perempuan     84
4      Eko  Laki-laki     65
5    Fitri  Perempuan     60
6     Gita  Perempuan     63
7   Hafidz  Laki-laki     75
8    Ihsan  Laki-laki     75
9      Iva  Perempuan     74
10  Kurnia  Perempuan     39
11  Lukman  Laki-laki     75
12    Maya  Perempuan     69
13    Tika  Perempuan     73
14    Wati  Perempuan     76


Dari output diatas dapat dilihat nomor data telah terurut dari 0 dan seterusnya. 

Mari kita mencoba untuk mengurutkan data menurut nilai dari tertinggi sampai yang terendah

In [11]:
data = data.sort_values("Nilai", ascending=False)
data = data.reset_index(drop=True)
print(data)

      Nama    Kelamin  Nilai
0    April  Perempuan     90
1      Eka  Perempuan     84
2     Andi  Laki-laki     80
3     Dewi  Perempuan     76
4     Wati  Perempuan     76
5   Hafidz  Laki-laki     75
6    Ihsan  Laki-laki     75
7   Lukman  Laki-laki     75
8      Iva  Perempuan     74
9     Tika  Perempuan     73
10    Maya  Perempuan     69
11     Eko  Laki-laki     65
12    Gita  Perempuan     63
13   Fitri  Perempuan     60
14  Kurnia  Perempuan     39


Dari ouput diatas diketahui bahwa nilai tertinggi April (Perempuan) dengan nilai 90, disusul dengan Eka (Perempuan) dengan nilai 84, kemudian diposisi tiga disusul oleh Andi (Laki-laki) dengan nilai 80 dan seterusnya dapat dilihat dari output diatas. 

##### Memfilter baris dan membuat kolom

Menampilkan data dengan 1 kolom. Dalam hal ini hanya menampilkan data variabel "Nama"

In [12]:
data['Nama']

0      April
1        Eka
2       Andi
3       Dewi
4       Wati
5     Hafidz
6      Ihsan
7     Lukman
8        Iva
9       Tika
10      Maya
11       Eko
12      Gita
13     Fitri
14    Kurnia
Name: Nama, dtype: object

Sebelum kita melanjutkan pembahasan selanjutnya, mari kita bahas Operator penghubung terlebih dahulu.

Operator penghubung 
- $==$ (sama dengan)
- $!=$ (tidak sama dengan)
- $<$ (kurang dari)
- $>$ (lebih besar dari)
- $<=$ (kurang dari atau sama dengan)
- $>=$ (lebih besar atau sama dengan)

Mencari data dari variabel "Nama" dengan nama "Iva"

In [13]:
data['Nama'] == "Iva"

0     False
1     False
2     False
3     False
4     False
5     False
6     False
7     False
8      True
9     False
10    False
11    False
12    False
13    False
14    False
Name: Nama, dtype: bool

Dari output diatas akan menampilkan data dengan tipe data boolean. Perlu diketahui bahwa tipe data boolean merupakan tipe data dengan keluaran True atau False. Dari data diatas data dengan Nama Iva terdapat pada data baris ke-8 (ditunjukkan dengan "True") sedangkan data lainnya False (data bukan dengan Nama Iva). 

Selanjutnya, mari kita coba filtering data

In [14]:
data[data['Nama']=="Gita"]

Nama    Kelamin  Nilai
12  Gita  Perempuan     63

Perintah diatas ingin mengetahui data dengan Nama Gita. Didapatkan Gita berjenis kelamin perempuan dan Nilainya yaitu 63. 

Menampilkan data dengan Nilai kurang dari 80. 

In [15]:
data[data['Nilai']<80]

Nama    Kelamin  Nilai
3     Dewi  Perempuan     76
4     Wati  Perempuan     76
5   Hafidz  Laki-laki     75
6    Ihsan  Laki-laki     75
7   Lukman  Laki-laki     75
8      Iva  Perempuan     74
9     Tika  Perempuan     73
10    Maya  Perempuan     69
11     Eko  Laki-laki     65
12    Gita  Perempuan     63
13   Fitri  Perempuan     60
14  Kurnia  Perempuan     39

Untuk memudahkan jumlah data dengan Nilai kurang dari 80. Mari kita mengubah nomor baris data 

In [16]:
data[data['Nilai']<80].reset_index(drop=True)

Nama    Kelamin  Nilai
0     Dewi  Perempuan     76
1     Wati  Perempuan     76
2   Hafidz  Laki-laki     75
3    Ihsan  Laki-laki     75
4   Lukman  Laki-laki     75
5      Iva  Perempuan     74
6     Tika  Perempuan     73
7     Maya  Perempuan     69
8      Eko  Laki-laki     65
9     Gita  Perempuan     63
10   Fitri  Perempuan     60
11  Kurnia  Perempuan     39

Pengurutan nomor data dengan menggunakan reset_index. Didapatkan Nilai kurang dari 80 sebanyak 12 orang. 

##### Membuat kolom baru

Operator Matematika
- $+$ (Tambah)
- $-$ (Kurang)
- $*$ (Kali)
- $/$ (Bagi)

Menambah kolom baru dengan nama "Nilai_akhir" dari variabel "Nilai" + 10

In [17]:
data['Nilai_akhir'] = data['Nilai']+10

In [18]:
data

Nama    Kelamin  Nilai  Nilai_akhir
0    April  Perempuan     90          100
1      Eka  Perempuan     84           94
2     Andi  Laki-laki     80           90
3     Dewi  Perempuan     76           86
4     Wati  Perempuan     76           86
5   Hafidz  Laki-laki     75           85
6    Ihsan  Laki-laki     75           85
7   Lukman  Laki-laki     75           85
8      Iva  Perempuan     74           84
9     Tika  Perempuan     73           83
10    Maya  Perempuan     69           79
11     Eko  Laki-laki     65           75
12    Gita  Perempuan     63           73
13   Fitri  Perempuan     60           70
14  Kurnia  Perempuan     39           49

Dapat dilihat pada kolom terakhir terdapat penambahan kolom dengan nama "Nilai_akhir". 

##### Menambahkan baris data

Penginputan data baru seperti syntax berikut

In [27]:
data1 = [{'Nama' : "Didi", 'Kelamin' : "Laki-laki", 'Nilai' : 89, 'Nilai_akhir' : 89+10}]
data1

[{'Nama': 'Didi', 'Kelamin': 'Laki-laki', 'Nilai': 89, 'Nilai_akhir': 99}]

Dapat dilihat pada data diatas memiliki informasi Nama Didi, Jenis Kelamin Laki-laki dan Nilai 89, Nilai akhir 89+10.

Membentuk dataframe

In [28]:
baru1 = pd.DataFrame(data1)
baru1

Kelamin  Nama  Nilai  Nilai_akhir
0  Laki-laki  Didi     89           99

Penggabungkan data awal dengan data baru dengan data.append()

In [44]:
data = data.append(baru1).reset_index(drop=True)
data

Kelamin    Nama  Nilai  Nilai_akhir
0   Perempuan   April     90          100
1   Perempuan     Eka     84           94
2   Laki-laki    Andi     80           90
3   Perempuan    Dewi     76           86
4   Perempuan    Wati     76           86
5   Laki-laki  Hafidz     75           85
6   Laki-laki   Ihsan     75           85
7   Laki-laki  Lukman     75           85
8   Perempuan     Iva     74           84
9   Perempuan    Tika     73           83
10  Perempuan    Maya     69           79
11  Laki-laki     Eko     65           75
12  Perempuan    Gita     63           73
13  Perempuan   Fitri     60           70
14  Perempuan  Kurnia     39           49
15  Laki-laki    Didi     89           99
16  Laki-laki    Didi     89           99

## Daftar Pustaka

https://medium.com/codelabs-unikom/ketika-python-bertemu-pandas-part-1-993bc9122f13